siphon/ncss.py

```diff
@@ -341,15 +344,33 @@ def parse_xml_dataset(elem, handle_units):
     @response_handlers.register('application/x-netcdf')
     @response_handlers.register('application/x-netcdf4')
     def read_netcdf(data, handle_units):  # pylint:disable=unused-argument
-        with NamedTemporaryFile() as tmp_file:
-            tmp_file.write(data)
-            tmp_file.flush()
-            return Dataset(tmp_file.name, 'r')
+        osType = platform.architecture()
+        if osType[1].lower() == 'windowspe':
+            with NamedTemporaryFile(delete=False) as tmp_file:
+                tmp_file.write(data)
+                tmp_file.flush()
+                atexit.register(deletetempfile,tmp_file.name)
+                return Dataset(tmp_file.name, 'r')
+        else:
+            with NamedTemporaryFile() as tmp_file:
+                tmp_file.write(data)
+                tmp_file.flush()
+                return Dataset(tmp_file.name, 'r')
+
 except ImportError:
     import warnings
     warnings.warn('netCDF4 module not installed. '
                   'Will be unable to handle NetCDF returns from NCSS.')
 
+def deletetempfile(fname):
+    try:
+        remove(fname)
+    except PermissionError:
```

Comment:
`PermissionError` it turns out is >= Python 3.3. I don't mind dropping Python 3.2, but I'm not quite prepared to drop Python 2.7. Since `PermissionError` is a subclass of `OSError`, can you use that instead?